lib/LanguageServerRename/Model/NodeUtils.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerRename\Model;
+
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\ArrayElement;
+use Microsoft\PhpParser\Node\ClassConstDeclaration;
+use Microsoft\PhpParser\Node\ConstElement;
+use Microsoft\PhpParser\Node\Expression\MemberAccessExpression;
+use Microsoft\PhpParser\Node\Expression\ScopedPropertyAccessExpression;
+use Microsoft\PhpParser\Node\Expression\Variable;
+use Microsoft\PhpParser\Node\MethodDeclaration;
+use Microsoft\PhpParser\Node\Parameter;
+use Microsoft\PhpParser\Node\PropertyDeclaration;
+use Microsoft\PhpParser\Node\QualifiedName;
+use Microsoft\PhpParser\Node\Statement\ClassDeclaration;
+use Microsoft\PhpParser\Node\Statement\ForeachStatement;
+use Microsoft\PhpParser\Node\Statement\InterfaceDeclaration;
+use Microsoft\PhpParser\Node\Statement\TraitDeclaration;
+use Microsoft\PhpParser\Node\StringLiteral;
+use Microsoft\PhpParser\Token;
+use Phpactor\Extension\LanguageServerBridge\Converter\PositionConverter;
+use Phpactor\LanguageServerProtocol\Position;
+use Phpactor\LanguageServerProtocol\Range;
+
+class NodeUtils
+{
+    public function getNodeNameText(Node $node, ?string $name = null): ?string
+    {
+        $token = $this->getNodeNameToken($node, $name);
+        if ($token === null) {
+            return null;
+        }
+        $text = (string)$token->getText($node->getFileContents());
+        if (mb_substr($text, 0, 1) == '$') {
```

Comment:
As methioned before this is specific to a certain class of rename, and should not be generic.

Reply:
That is the old class. Look at `NodeUtils2`.